### DISCUSSION TITLE
1 VirtualFolder that writes to 2 VFS backends

### DISCUSSION THREAD

**Discussion by Smithx10** [drakkan/sftpgo#540] [Ideas]
Good Day,

Some requirements in an organization may require writing data to Azure and S3.  Would we be open creating a type of Virtual Folder that only succeeds when it has written the same file to 2 different backends?

Thoughts?  

Thank You,
Bruce Smith

**Comment by drakkan** [comment]
Hi,

I'm not sure about this. SFTPGo is not a sync solution. Which backend should be used to list/read files? We cannot guarantee consistency: if one upload is successful and the other fails, we may not be able to remove the uploaded file.

I'm open to discussion but my first thought is that if you need to backup/sync your data you should use a sync/backup solution and not an SFTP/FTP server

**Comment by Smithx10** [comment]
@drakkan,

I was also contemplating those scenarios.  

The other issue I was thinking about is how to report when a 1 of the 2 files has been deleted outside of SFTP.    The consumer of SFTP will have no idea, and only the operator.  

I also tend to like pushing the behavior of applications and state onto developers so they can better understand what is happening.  

I agree.  Probably should be handled by the client.

Thanks for the thoughts,
Bruce
